docs/SoftwareCapabilities.md

```diff
@@ -0,0 +1,96 @@
+=====================================
+
+CHERIoT is a capability system.
+This extends from the hardware (all memory accesses require an authorising capability in a register, provided as an operand to an instruction) up through the software abstractions in the RTOS.
+In a capability system, any privileged operation requires an explicit capability to authorise it.
+
+Privilege is a somewhat fluid notion in a system with fine-grained mutual distrust but, in general, any operation that affects state beyond the current compartment is considered privileged.
+This includes actions such as:
+
+ - Acknowledging interrupts
+ - Allocating heap memory
+ - Establishing network connections
+
+Each of these should require that the caller present a capability that authorises the callee to perform the action on behalf of the caller.
+
+Delegation is an important part of a capability system.
+A capability may be passed from compartment A to compartment B, which can then use it to ask C to perform some privileged operation.
+The identity of the immediate caller does not matter.
+
+CHERI capabilities to software-defined capabilities
+---------------------------------------------------
+
+CHERI provides a hardware mechanism for building software-defined capabilities: sealing.
+The sealing mechanism allows a pointer (a CHERI capability) to be made immutable and unusable until it is unsealed using an authorising capability.
+The CHERIoT ISA has a limited set of space for sealing types and so these are virtualised with the allocator's [token](../sdk/include/token.h) APIs.
+These APIs combine allocation and sealing, returning both sealed and unsealed capabilities to an object.
+The unsealed capability can be used directly, the sealed capability can be passed to other code and unsealed only by calling the allocator's `token_unseal` function with the capability used to allocate the object.
+
+A compartment can use this to generate software-defined capabilities that represent dynamic resources.
+For example, a network stack can use it to allocate the state associated with a connection.
+The scheduler uses the same mechanism for providing capabilities for cross-thread communication so that, for example, only a holder of the relevant capability can send or receive messages in a message queue.
+
+Static software-defined capabilities
+------------------------------------
+
+In addition to the dynamic software-defined capabilities, it is often useful to provision a set of capabilities to a compartment at build time.
+The simplest case for this is (allocator capabilities)[Allocator.md], which authorise allocation and so are necessary to be able to create any dynamic software-defined capabilities.
+The scheduler also uses this mechanism for capabilities that allow interaction with [interrupts](Interrupts.md).
+
+Objects created in this way are allocated outside of a compartment's global region but are accessed only via capabilities provided by the loader.
```

Comment:
```suggestion
Objects created in this way are allocated outside of any compartment's global region and are accessed only via capabilities provided by the loader.
The initial contents of these objects are specified during definition, which is often located within a compartment that does not have the rights to unseal the resulting object.
Therefore, these contents, as well as the type of the software-defined capability, are reported in the linker's audit summary and must be validated; see below.
```

Reply:
If that's too many words, it would probably still be good to forward reference the auditing section.